generate_cache.py

```diff
@@ -108,11 +108,17 @@ def write_cache_entries(name, specs, hash_stacks):
             os.makedirs(package_dir)
         spec_details = []
         for i, spec in enumerate(speclist):
+            try:
+                _ = spec.compiler
+                compiler_str = spec.format("{compiler.name}@{compiler.version}")
+            except AttributeError:
+                compiler_str = "no-compiler"
```

Comment:
Would `none` be a better name? I am trying to think about how this would look/be queried on the website.

Reply:
I'm fine with `none`